app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdEffectHandler.kt

```diff
@@ -26,8 +29,23 @@ class ScanSimpleIdEffectHandler @AssistedInject constructor(
       .addConsumer(OpenPatientShortCodeSearch::class.java, { uiActions.openPatientShortCodeSearch(it.shortCode) }, schedulersProvider.ui())
       .addConsumer(OpenPatientSummary::class.java, { uiActions.openPatientSummary(it.patientUuid) }, schedulersProvider.ui())
       .addConsumer(OpenAddIdToPatientScreen::class.java, { uiActions.openAddIdToPatientScreen(it.identifier) }, schedulersProvider.ui())
+      .addTransformer(SearchPatient::class.java, searchPatient())
       .build()
 
+  private fun searchPatient(): ObservableTransformer<SearchPatient, ScanSimpleIdEvent> {
+    return ObservableTransformer { effects ->
+      val scannedUuid = effects
+          .map { it.scannedUuid }
+          .share()
+
+      effects
+          .observeOn(schedulersProvider.io())
+          .flatMap { patientRepository.findPatientWithBusinessId(it.scannedUuid.toString()) }
+          .withLatestFrom(scannedUuid)
+          .map { (patient, scannedUuid) -> PatientSearchCompleted(patient, scannedUuid) }
```

Comment:
I tried to get rid of the `.share()` operator usage here and make this into a single chain.

```kotlin
effects
          .switchMap { effect ->
            val scannedUuid = effect.scannedUuid

            patientRepository
                .findPatientWithBusinessId(scannedUuid.toString())
                .subscribeOn(schedulersProvider.io())
                .map { PatientSearchCompleted(it, scannedUuid) }
          }
```

What do you think about using this instead?

Reply:
Yeah, this works 👍🏾